### PR TITLE
Adds Modules to manage IIS

### PR DESCRIPTION
**Pull request by henrikwallstrom**
Wraps the Web Server Administration module for PowerShell into Ansible modules.

### CODE REVIEW

**Comment by bcoca** [comment]
@cchurch

**Comment by BenjaminVanRyseghem** [comment]
:+1:

**Comment by henrikwallstrom** [comment]
@cchurch is this something that we/you want in ansible-module-extra?

**Comment by jhawkesworth** [comment]
Have not had the chance to test this yet but I am sure others would be
interested in this functionality.

Jon

**Comment by henrikwallstrom** [comment]
:+1:  let me know if there is something that should be done differently when you have had time to test

File: windows/win_iis_webapplication.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      - State of the web application
+    choices:
+      - present
+      - absent
+    required: false
+    default: null
+    aliases: []
+  physical_path:
+    description:
+      - The physical path on the remote host to use for the new applicatiojn. The specified folder must already exist.
```
Comment (jhawkesworth) [inline windows/win_iis_webapplication.py:52]: possible typo: applicatiojn - > application

**Comment by gregdek** [comment]
Hey @jhawkesworth -- aside from the typo, does this module pass review? Because if so, I'm happy to ping other folks to get that second +1 review we need to merge.

**Comment by smadam813** [comment]
Heck, I was just about to write this.  I can review this code too.

**Comment by gregdek** [comment]
@smadam813 That would be great! A +1 from you would get this merged.

**Comment by smadam813** [comment]
@whenrik Is there an assumption that the server will have IIS already installed as well as the powershell IIS tools to use cmdlets such as New-Website?  So far it looks good and is working properly as long as I have the dependencies installed.  There are some typos in the py files, confusing variable names at times, and missing descriptions / examples, but it's not really a big deal.  I will continue testing tomorrow.

**Comment by smadam813** [comment]
@gregdek @whenrik Other than the fact I needed to install IIS features manually within Windows before being able to use this module, the error handing works well, and every module did what it was supposed to.  You get my +1 as this will be a very useful module.

**Comment by gregdek** [comment]
Merging. @smadam813 thanks for the review, and @whenrik thanks for the code!

**Comment by abadger** [comment]
Just a note that I fixed up quite a few docs bugs here this morning: https://github.com/ansible/ansible-modules-extras/commit/dec7d95d514ca89c2784b63d836dd6fb872bdd9c

Understandable that there will be a few issues when adding so many new modules :-)  One thing a reviewer or author can do to check the documentation before commit is to run the documentation build.  From a checkout of the source with the changes applied:

```
$ cd docsite
$ make docs
```

**Pull request merged**